docs/src/pages/components/data-grid/getting-started/migration-v4.md

```diff
@@ -0,0 +1,432 @@
+      ) {
+        prevRenderContext.current = renderContext;
+        const params = {
+          firstRowIndex: renderContext.firstRowIndex,
+          lastRowIndex: renderContext.lastRowIndex,
+        };
+      }
+    });
+  }, [apiRef]);
+
+  <DataGridPro apiRef={apiRef} />;
+  ```
+
+### Properties of the Grid
+
+- Some event listeners and DOM attributes were removed from `GridCell` and `GridRow` to improve performances.
+
+  The following props were removed.
+
+  - `onCellBlur`
+  - `onCellOver`
+  - `onCellOut`
+  - `onCellEnter`
+  - `onCellLeave`
+  - `onRowOver`
+  - `onRowOut`
+  - `onRowEnter`
+  - `onRowLeave`
+
+  If you depend on them, you can use `componentsProps.row` and `componentsProps.cell` to pass custom props to the row or cell.
+  For more information, check [this page](https://mui.com/components/data-grid/components/#row). Example:
+
+  ```diff
+  -<DataGrid onRowOver={handleRowOver} />;
+  +<DataGrid
+  +  componentsProps={{
+  +    row: { onMouseOver: handleRowOver },
+  +  }}
+  +/>;
+  ```
+
+  The `data-rowindex` and `data-rowselected` attributes were removed from the cell element. Equivalent attributes can be found in the row element.
+
+  The `data-editable` attribute was removed from the cell element. Use the `.MuiDataGrid-cell--editable` CSS class.
+
+  The `data-mode` attribute was removed from the cell element. Use the `.MuiDataGrid-cell--editing` CSS class.
+
+### State access
+
+- The state direct access is not considered part of the public API anymore. We only guarantee that the selectors continue to work between minor releases.
+  We advise you to avoid accessing directly a state sub-key and to use selectors or `apiRef` methods whenever possible.
+  You can replace the following state access by there matching selectors:
+
+  | Direct state access     | Selector                           |
+  | ----------------------- | ---------------------------------- |
+  | `state.rows`            | `gridRowsStateSelector`            |
+  | `state.filter`          | `gridFilterStateSelector`          |
+  | `state.sorting`         | `gridSortingStateSelector`         |
+  | `state.editRows`        | `gridEditRowsStateSelector`        |
+  | `state.pagination`      | `gridPaginationSelector`           |
+  | `state.columns`         | `gridColumnsSelector`              |
+  | `state.columnMenu`      | `gridColumnMenuSelector`           |
+  | `state.focus`           | `gridFocusStateSelector`           |
+  | `state.tabIndex`        | `gridTabIndexStateSelector`        |
+  | `state.selection`       | `gridSelectionStateSelector`       |
+  | `state.preferencePanel` | `gridPreferencePanelStateSelector` |
+  | `state.density`         | `gridDensitySelector`              |
+  | `state.columnReorder`   | `gridColumnReorderSelector`        |
+  | `state.columnResize`    | `gridColumnResizeSelector`         |
+
+- The `apiRef.current.getState` method was removed. You can directly access the state through `apiRef.current.state`
```

Comment:
Strange to explain how to access state just after explaining it is not the good way to access data
```suggestion
- Since state should be accessed by selectors, the `apiRef.current.getState` method was removed. You can still directly access the state through `apiRef.current.state` (not recommended)
```

Reply:
You have to access the state object to give it to selectors.
For instance 

```tsx
const pageSize = gridPageSizeSelector(apiRef.current.state)
```

What we want to avoid is people doing

```tsx
const pageSize = apiRef.current.state.pagination.pageSize
```